File: src/RehabEasy.Domain/Models/PatientClinicalNoteHistoryEntry.cs
namespace RehabEasy.Domain.Models;

public sealed class PatientClinicalNoteHistoryEntry
{
    public long Id { get; init; }
    public string PatientId { get; init; } = string.Empty;
    public string Content { get; init; } = string.Empty;
    public DateTimeOffset SavedAt { get; init; }
}

File: java/src/main/java/com/rehabeasy/model/PatientClinicalNoteHistoryEntry.java
package com.rehabeasy.model;

import java.time.Instant;

public record PatientClinicalNoteHistoryEntry(
        long id,
        String patientId,
        String content,
        Instant savedAt
) {
    public PatientClinicalNoteHistoryEntry {
        patientId = patientId == null ? "" : patientId;
        content = content == null ? "" : content;
        savedAt = savedAt == null ? Instant.EPOCH : savedAt;
    }
}
